gptel-bedrock.el

```diff
@@ -528,7 +549,8 @@ Convenient to use with `cl-multiple-value-bind'"
     (cond
       ((and key-id secret-key) (cl-values key-id secret-key token))
       ((and profile) (gptel-bedrock--fetch-aws-profile-credentials profile))
-      (t (user-error "Missing AWS credentials; currently only environment variables are supported")))))
+      ((gptel-bedrock--fetch-instance-credentials))
```

Comment:
This runs on every request.  `call-process` is synchronous and blocks Emacs.  I would like gptel to _never_ block Emacs.

Making this asynchronous in the gptel-request pipeline is more work, and probably best avoided for now.

If these credentials can be cached, would you consider caching them instead?  Then it will block the first time but subsequent requests should be processed instantaneously.